lib/node/runner/checker_ballot_transaction.go

```diff
@@ -244,14 +254,79 @@ func ValidateOp(st *storage.LevelDBBackend, source *block.BlockAccount, op trans
 			err = errors.ErrorFrozenAccountNoDeposit
 			return
 		}
-		// If it's a frozen account, everything must be withdrawn
 		if source.Linked != "" {
+			// Source account's linked address must be equal to target address
+			if source.Linked != casted.TargetAddress() {
+				err = errors.ErrorUnfreezingToInvalidLinkedAccount
+			}
+			// If it's a frozen account, everything must be withdrawn
 			var expected common.Amount
 			expected, err = source.Balance.Sub(common.BaseFee)
 			if casted.Amount != expected {
 				err = errors.ErrorFrozenAccountMustWithdrawEverything
 				return
 			}
+			// Unfreezing must be done after X period from unfreezing request
+			var saved []block.BlockOperation
+			iterFunc, closeFunc := block.GetBlockOperationsBySource(st, source.Address, nil)
+
+			for {
+				t, hasNext, _ := iterFunc()
+				if !hasNext {
+					break
+				}
+				saved = append(saved, t)
+			}
+			closeFunc()
+
+			bo := saved[len(saved)-1]
+			var lastblock block.Block
+			lastblock, err = block.GetLatestBlock(st)
+			if err != nil {
+				return
+			}
+			if bo.Type != transaction.OperationUnfreezingRequest {
+				err = errors.ErrorUnfreezingRequestNotRequested
+				return
+			}
+
+			// unfreezing period is originally 241920(=12*60*24*14), but here is set to 4 for the test.
+			if lastblock.Height-bo.BlockHeight <= uint64(4) {
```

Comment:
That must be a constant or variable for testing purpose, but not here in the code.

Reply:
@Geod24 Ok I'll make a unfreezing period constant in constant.go.